python/whylogs/core/validators/condition_validator.py

```diff
@@ -41,11 +81,30 @@ def columnar_validate(self, data: PreprocessedColumn) -> None:
                 if not valid:
                     self.failures[cond_name] += 1
                     count_failures += 1
+                    if self.enable_sampling and validate_with_row_id and isinstance(self._sampler, ds.var_opt_sketch):
+                        self._sampler.update(identity_values[index])
+                    # if we don't have an identity column, sample the validated values
+                    elif self.enable_sampling and isinstance(self._sampler, ds.var_opt_sketch):
+                        self._sampler.update(x)
                     for action in self.actions:
-                        action(self.name, cond_name, x)
+                        if validate_with_row_id:
+                            action(self.name, cond_name, x, identity_values[index])
+                        else:
+                            action(self.name, cond_name, x)  # type: ignore
 
         self.total = count
 
+    def sample_failed_conditions(self) -> List[Any]:
```

Comment:
```suggestion
    def get_samples(self) -> List[Any]:
```

To match the sketch's name. We aren't actively computing the sample with this, just reading it out of the sketch.